### DISCUSSION TITLE
Support for distributed package management

### DISCUSSION THREAD

**Discussion by inie0722** [microsoft/vcpkg#25329] [Ideas]
In places where the network environment is not very smooth, the centralized download speed may be very slow. For example, when trying to download GitHub files in mainland China, it is extremely slow without a proxy. If you build distributed package management based on ipfs may be able to improve these situations.

**Comment by frank-dspeed** [comment]
? out of my view that is already supported google vcpkg registries you can use for example with ipfs the git registry.

**Comment by inie0722** [comment]
> ? out of my view that is already supported google vcpkg registries you can use for example with ipfs the git registry.

There are some differences between [ipfs](https://ipfs.io/) and normal file systems.
I expect it to be as easy and convenient as  [npm-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs).

**Comment by AugP** [comment]
This is an interesting suggestion, I added it to our backlog.

**Comment by frank-dspeed** [comment]
https://docs.ipfs.io/how-to/host-git-style-repo/

i do not know whats not covered by that if we can do something on the ipfs side i would be free to integrate that but out of my view ipfs got enough connect ability in enough formats that are supported by vcpkg already.

**Comment by frank-dspeed** [comment]
also torrent is useable and fits the use case you simply create a torrent folder and a torrent then your ready

**Comment by inie0722** [comment]
yes, vcpkg already supports ipfs.
I think we should maintain a vcpkg-ipfs(vcpkg source on ipfs), automatically sync to ipfs when a new package is released.
Then use `vcpkg-ipfs install xxx`, you can easily install the package.

**Reply by frank-dspeed** [reply]
but i do not understand ipfs is fuse mountable so why do we not use the filesystem with vcpkg?

**Reply by frank-dspeed** [reply]
ipfs supports http via gateway so vcpkg works with that also

**Reply by inie0722** [reply]
> ipfs supports http via gateway so vcpkg works with that also

yes we can use http gateway.
I mean we can generate an ipfs registry for the default registries.
This way we don't have to write our own ipfs registries.

**Reply by inie0722** [reply]
Create a public default ipfs registries.
As for the download method, you can use http gateway or fuse mountable.
